main.tf

```diff
@@ -45,7 +45,10 @@ resource "azurerm_app_service" "main" {
     }
 
     app_settings = {
+    "BOARD_ID" = "Board123"
```

Comment:
Here you're using a hardcoded string (`"Board123"`) rather than the variable, so the BOARD_ID variable never overrides this